gmso/formats/prm_writer.py

```diff
@@ -0,0 +1,124 @@
+                        site.atom_type.mass, n_decimals=6, name="mass"
+                    ),
+                )
+            )
+        for atom in unique_atoms:
+            f.write("MASS -1 {:8s} {:8s}\n".format(atom[0], atom[1]))
+
+        # TODO: Works for harmonic bonds
+        f.write("\nBONDS\n")
+        for bond in topology.bond_types:
+            atom1, atom2 = bond.member_types
+            f.write(
+                "{:8s} {:8s} {:.5f} {:.5f}\n".format(
+                    atom1, atom2, bond.parameters["k"], bond.parameters["r_eq"]
+                )
+            )
+
+        f.write("\nANGLES\n")
+        for angle in topology.angle_types:
+            atom1, atom2, atom3 = angle.member_types
+            if angle.name == "UreyBradleyAnglePotential":
+                f.write(
+                    "{:8s} {:8s} {:8s} {:.5f} {:.5f} {:.5f} {:.5f}\n".format(
+                        atom1,
+                        atom2,
+                        atom3,
+                        angle.parameters["k"],
+                        angle.parameters["theta_eq"],
+                        angle.parameters["kub"],
+                        angle.parameters["r_eq"],
+                    )
+                )
+            else:  # assume harmonic style:
+                f.write(
+                    "{:8s} {:8s} {:8s} {:.5f} {:.5f}\n".format(
+                        atom1,
+                        atom2,
+                        atom3,
+                        angle.parameters["k"],
+                        angle.parameters["theta_eq"],
+                    )
+                )
+
+        # These dihedrals need to be PeriodicTorsion Style (Charmm style)
+        f.write("\nDIHEDRALS\n")
+        for dihedral in topology.dihedral_types:
+            # works for PeriodicTorsion
+            atom1, atom2, atom3, atom4 = dihedral.member_types
+            f.write(
+                "{:8s} {:8s} {:8s} {:8s} {:.5f} {:5f} {:.5f}\n".format(
+                    atom1,
+                    atom2,
+                    atom3,
+                    atom4,
+                    dihedral.parameters["k"][0],
+                    dihedral.parameters["n"][0],
+                    dihedral.parameters["phi_eq"][0],
+                )
+            )
+
+        # TODO: No support for harmonic impropers
+
+        f.write("\nIMPROPER\n")
+        for improper in topology.improper_types:
+            atom1, atom2, atom3, atom4 = improper.member_types
+            f.write(
+                "{:8s} {:8s} {:8s} {:8s} {:.5f} {:5f} {:.5f}\n".format(
+                    atom1,
+                    atom2,
+                    atom3,
+                    atom4,
+                    improper.parameters["phi_k"][0],
+                    improper.parameters["n"][0],
+                    improper.parameters["phi_eq"][0],
+                )
+            )
+
+        f.write("\nNONBONDED\n")
+        nonbonded14 = topology.scaling_factors[0][2]
```

Comment:
This doesn't need to be fixed here, but I feel like this attribute should be more readable rather than knowing that 0th index is LJ and 1st index is electrostatics. Maybe a dict? I can open a separate issue.

Reply:
Was thinking the same thing myself. I think it should also be {"electrostatics":{"1-4":0.5}}, instead of a fixed list. Could potentially be some odd 1-5 interactions at some point.

Reply:
Yeah, that's sort of what I was thinking as well.